client/js/contextMenuFactory.js

```diff
@@ -0,0 +1,204 @@
+"use strict";
+const $ = require("jquery");
+const socket = require("./socket");
+const utils = require("./utils");
+const JoinChannel = require("./join-channel");
+const ContextMenu = require("./contextMenu");
+const contextMenuActions = [];
+const contextMenuItems = [];
+
+module.exports = {
+	addContextMenuItem,
+	createContextMenu,
+};
+
+addDefaultItems();
+
+/**
+ *
+ * @param opts
+ * @param {function(Object)} [opts.check] - Function to check whether item should show on the context menu, called with the target jquery element, shows if return is truthy
+ * @param {string|function(Object)} opts.className - class name for the menu item, should be prefixed for non-default menu items (if function, called with jquery element, and uses return value)
+ * @param {string|function(Object)} opts.data - data that will be sent to the callback function (if function, called with jquery element, and uses return value)
+ * @param {string|function(Object)} opts.display - text to display on the menu item (if function, called with jquery element, and uses return value)
```

Comment:
I often see `displayName` as a wording to call this kind of thing (`label` too). Nitpicky and don't feel like I expect you to change, I just wanted to put this out there in case you were having second thoughts.
(For a second, I wondered if this would be `display` as in `display: none` and thought it could be confusing, but that might be too far fetched)

Reply:
Happy to change this.